src/lambda-handler.ts

```diff
@@ -40,43 +40,61 @@ const streamify_handler: StreamifyHandler = async ( event, response ) => {
 		delete args.presign;
 	}
 
-	let s3_response = await getS3File( config, key, args );
-	if ( ! s3_response.Body ) {
-		throw new Error( 'No body in file.' );
-	}
-	let buffer = Buffer.from( await s3_response.Body.transformToByteArray() );
+	try {
```

Comment:
There seems to be too much things going on here. Instead of only checking the exception we expect from `getS3File()` we're now checking for an exception from a lot of other things, including one that we throw based on `s3_response.Body`. Let's try and simplify this and return/end early if there's an AccessDenied error:

```
try {
    s3_response = await getS3File( config, key, args );
} catch ( e: any ) {
    if ( e.Code === 'AccessDenied' ) {
        // set status, content-type, write
        response.end();
        return;
    }
    throw e;
}

// if ( ! s3_response.Body ) { ...
```

Or something along these lines. This will also mean that the rest of the code remains unchanged and the exception handling is much, much closer to the line we expect to produce that exception.